addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/content/MetadataCacheKey.java

```diff
@@ -0,0 +1,65 @@
+package org.commonjava.indy.pkg.maven.content;
+
+import org.commonjava.indy.model.core.StoreKey;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+public final class MetadataCacheKey
+                implements Serializable
+{
+    private static final long serialVersionUID = 1L;
+
+    private final StoreKey storeKey;
+
+    private final String path;
+
+    public MetadataCacheKey( StoreKey storeKey, String path )
+    {
+        this.storeKey = storeKey;
+        this.path = path;
+    }
+
+    public StoreKey getStoreKey()
+    {
+        return storeKey;
+    }
+
+    public String getPath()
+    {
+        return path;
+    }
+
+    @Override
+    public boolean equals( Object o )
+    {
+        if ( this == o )
+            return true;
+        if ( o == null || getClass() != o.getClass() )
+            return false;
+        MetadataCacheKey that = (MetadataCacheKey) o;
+        return storeKey.equals( that.storeKey ) && path.equals( that.path );
```

Comment:
Objects.equals() instead to avoid storeKey null or path null?

Reply:
Will do.